### ISSUE TITLE
formaction not respected on buttons/input

### ISSUE THREAD

**Issue by boxxxie** [unpoly/unpoly#173]
solution that i have working in my app, i know that there is some other way to find a form in unpoly, but i haven't tried that.
one nice thing about this solution is that i can use it on any tag, but without JS non input/buttons wont work.

```javascript
up.compiler('[formaction]', function(element, data) {
  const form = find_form(element) || document.querySelector(element.getAttribute('form'));

  const formaction = element.getAttribute('formaction');

  return up.on(element, "click", function(event){
    up.event.halt(event);
    if(form.checkValidity()){
      up.submit(form, {url: formaction});
    }
    else {
      form.reportValidity();
    }
  })
})
```

**Comment by adam12** [comment]
You're running v1 or pre-v1 correct? and not one of the v2 RC's?

**Comment by boxxxie** [comment]
yeah, not using v2.
i can't compile unpoly, so i'm using 1 version before v1. i just noticed the v1 was released.

if you guys can help me compile this project, then i can use v2 and post issues related to it.
i haven't been able to compile this project via coffee, and i didn't find instructions for compiling that worked for me (arch linux), also i've never used coffeescript before, so maybe i'm just missing something that is obvious to  people who work with it.

**Comment by adam12** [comment]
> if you guys can help me compile this project, then i can use v2 and post issues related to it.

You can get pre-build release candidates for V2 from [the V2 site](https://v2.unpoly.com/install), if you like. Tho do be aware that a) you might need the migrate script as well, and b) it's still a release candidate (tho I'm using it to much success already).

> i haven't been able to compile this project via coffee

Do you use VSCode by chance? and Docker? I use a DevContainer locally to build and test Unpoly, and I've been contemplating making it available in this repo, similar to how I did for `unpoly/site`.

**Comment by adam12** [comment]
I still believe this is fixed in v2, per [this thread](https://github.com/unpoly/unpoly/pull/147). Please confirm if you're able.

**Comment by boxxxie** [comment]
i use emacs. i use command line tools for building, usually `entr + fish or shell scrips` it's just easier to do that for me, as i have to support devs on other OSs and editors. i don't use docker for building, but i can see it's value as i've run into issues with versioned tools giving different output... for this type of thing i would prefer to build with near latest version of coffee and other tools, though.

i'll try the link you gave, but i imagine there is a branch for v2?

**Comment by boxxxie** [comment]
oh, seems like i brought this issue up before.... oh man, sorry about this.

anyway, i'm pretty interested in the docker file you have, so i can look at how you setup your system for compiling.

**Comment by adam12** [comment]
> for this type of thing i would prefer to build with near latest version of coffee

Not sure that's possible at this time, but in theory, if you could have the correct version of Ruby installed, a version of Node installed, and then the dependencies installed through the `Gemfile`, then it should work.

> i imagine there is a branch for v2

This is now the `master` branch (with v1 being in its own branch).

**Comment by adam12** [comment]
> i'm pretty interested in the docker file you have

Let me clean it up and I'll share it.

**Comment by boxxxie** [comment]
cool, thanks!

**Comment by adam12** [comment]
My branch for working inside Docker is here: https://github.com/adam12/unpoly/tree/devcontainer

See the [README](https://github.com/adam12/unpoly/tree/devcontainer/.devcontainer).

**Comment by triskweline** [comment]
Confirming that this is fixed in 2.0.

**Comment by adam12** [comment]
I think we're good to close this then.
